File: AccountServic/src/main/java/com/diary/AccountService/Entities/Account.java
package com.diary.AccountService.Entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@Table(name = "ACCOUNTS")
@Getter
@Setter
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private String acc_id;
    private String acc_number;
    private String acc_provider;
    private String acc_status;
    private String acc_type;
    private Long acc_balance;
    private Date created_at;
    private Date updated_at;
    private String user_id;

}

File: AccountService/src/Model/Account.ts
import { DataTypes, Model } from "sequelize";
import { sequalizeInstance } from "../config/dbconfig";

const Accounts = sequalizeInstance.define("accounts", {
  accId: {
    type: DataTypes.UUIDV4,
    autoIncrement: true,
    primaryKey: true,
  },
  acc_number: {
    type: DataTypes.STRING,
  },
  acc_provider: {
    type: DataTypes.STRING,
  },
  acc_status: {
    type: DataTypes.STRING,
  },
  acc_type: {
    type: DataTypes.STRING,
  },
  acc_balance: {
    type: DataTypes.DOUBLE,
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: new Date(),
  },
  updated_at: {
    type: DataTypes.DATE,
    defaultValue: new Date(),
  },
  user_id: {
    type: DataTypes.UUIDV4,
  },
});

export default Accounts;
